components/service/firefox-accounts/src/main/java/mozilla/components/service/fxa/sync/WorkManagerSyncManager.kt

```diff
@@ -436,19 +436,20 @@ class WorkManagerSyncWorker(
 
         // Process telemetry.
         syncResult.telemetry?.let {
-            // Yes, this is complete garbage.
+            // Yes, this is non-ideal...
             // But, what this does: individual 'process' function will report global sync errors
-            // as part of its corresponding ping. We don't want to report a global sync error twice,
+            // as part of its corresponding ping. We don't want to report a global sync error multiple times,
             // so we check for the boolean flag that indicates if this happened or not.
-            // There's a complete mismatch between what Glean supports and what we need
-            // it to do here. They don't support "nested metrics" and so we resort to these hacks.
+            // There's a complete mismatch between what Glean supports and what we need it to do here.
```

Comment:
This is a bit unfair, to be honest.

The sync team, the Glean teams and DS [designed and agreed](https://docs.google.com/document/d/1aJnqqsLnhIAzEJR4UylXO8jUVrkw68ZMGHMjKTANtQA/edit#heading=h.9qtbe4mxitxr) on the solution to send "one ping per engine sync". This is not, IMHO, a shortcoming of the SDK: this is doing what we all agreed on.

We can revisit the decision, of course.